kafka/kafka.go

```diff
@@ -271,25 +269,10 @@ func (t *subTopic) Subscribe(ctx context.Context, handler bps.Handler, options .
 	for {
 		select {
 		case <-ctx.Done():
```

Comment:
with stopping the subscription instead of cancelling the context you could just `return` here and `consumers.Wait()` in the `Stop()` function itself to block until the subscription is actually done